backend-api/src/functions/common/session/Session.ts

```diff
@@ -0,0 +1,61 @@
+import { AttributeValue } from "@aws-sdk/client-dynamodb";
+
+export enum SessionState {
+  AUTH_SESSION_CREATED = "ASYNC_AUTH_SESSION_CREATED",
+  BIOMETRIC_TOKEN_ISSUED = "ASYNC_BIOMETRIC_TOKEN_ISSUED",
+  BIOMETRIC_SESSION_FINISHED = "ASYNC_BIOMETRIC_SESSION_FINISHED",
+  RESULT_SENT = "ASYNC_RESULT_SENT",
+}
+
+type CommonSessionKeys = {
+  sessionId: string;
+  createdAt: number;
+  timeToLive: number;
+  clientId: string;
+  govukSigninJourneyId: string;
+  issuer: string;
+  clientState: string;
+  subjectIdentifier: string;
+  redirectUri?: string;
+};
+
+export type AuthSessionCreatedSession = CommonSessionKeys & {
+  sessionState: SessionState.AUTH_SESSION_CREATED;
+};
+
+export type BiometricTokenIssuedSession = CommonSessionKeys & {
+  sessionState: SessionState.BIOMETRIC_TOKEN_ISSUED;
+  documentType: string;
+  accessToken: string;
+  opaqueId: string;
+};
+
+export type BiometricSessionFinishedSession = CommonSessionKeys & {
+  sessionState: SessionState.BIOMETRIC_SESSION_FINISHED;
+  documentType: string;
+  accessToken: string;
+  opaqueId: string;
+  biometricSessionId: string;
+};
+
+export type ResultSentSession = CommonSessionKeys & {
+  sessionState: SessionState.RESULT_SENT;
+  documentType: string;
+  accessToken: string;
+  opaqueId: string;
+  biometricSessionId: string;
+};
+
+export type Session =
+  | AuthSessionCreatedSession
+  | BiometricTokenIssuedSession
+  | BiometricSessionFinishedSession
+  | ResultSentSession;
+
+export interface IUpdateSessionOperation {
```

Comment:
I'm just mulling (or marinadtng as Elishka would say) on this one. I like how the lambda decides what implementation of the operations to follow. 

Perhaps it's just the location of this interface! It would be good to think about partitioning (even if it's by folder structure) as much of the Dynamo implementation we can from the business logic/interfaces you've written that are so clean.

Reply:
Definitely happy to move the interface elsewhere. 

Something I considered was having the update operations just know about (and be able to tell you about) their valid 'FROM' session states and target session state, as well as what other fields need adding, in a dynamo-agnostic way. I just couldn't think of a neat way to work that list bit in in the time I'd taken on it. But will keep marinading.